hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/ReactiveConnection.java

```diff
@@ -50,6 +50,7 @@
 	CompletionStage<Result> select(String sql);
 	CompletionStage<Result> select(String sql, Object[] paramValues);
 	CompletionStage<ResultSet> selectJdbc(String sql, Object[] paramValues);
+	CompletionStage<ResultSet> selectJdbcOutsideTransaction(String sql, Object[] paramValues);
```

Comment:
Could you remind us why you need the additional method?

Reply:
@DavideD , since `SchemaMigrator` executes select queries on system tables and executes DDL statements (which update system tables), I assumed that these queries should be executed outside of a transaction (i.e., with `autocommit=true`). IIUC, this would help avoid contention with other work the DB is doing on schemas used by various applications.

If you look at the corresponding ORM code in [GenerationTargetToDatabase]
(https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/tool/schema/internal/exec/GenerationTargetToDatabase.java#L74-L85): you can see that the statement is created using a connection obtained from a `DdlTransactionIsolator`.

Javadoc for `DdlTransactionIsolator` says:
```
 * Provides access to a Connection that is isolated from
 * any "current transaction" with the designed purpose of
 * performing DDL commands
```
If you look at the `DdlTransactionIsolator` implementations, you will see that `autocommit` is set to `true` on the `Connection` that is obtained:
`DdlTransactionIsolatorJtaImpl`
`DdlTransactionIsolatorNonJtaImpl`
`DdlTransactionIsolatorProvidedConnectionImpl` relies on `JdbcConnectionAccessProvidedConnectionImpl` which sets `autocommit` to `true`

I didn't see any other way to execute a query outside of a transaction that returned a `CompletionStage<ResultSet>`, so I created a new method.

Is there some other method that should be used instead?

Reply:
I don't think we had this requirement before. I don't see any problem adding a new method.

But I mainly wanted a bit of an explanation written somewhere so that others know as well (maybe it wasn't obvious only to me though ;-)

At some point, we will need to add some javadoc to that class. Anyway, not something you need to worry about at the moment.

Reply:
Yes, I know this stuff is lacking in documentation. I will be adding that.